packages/amplify-e2e-tests/src/__tests__/uibuilder.test.ts

```diff
@@ -110,16 +110,21 @@ describe('amplify pull with uibuilder', () => {
       fs.readFileSync(path.join(__dirname, '..', 'cypress', 'uibuilder', 'uibuilder-spec.js')),
     );
 
-    const npmStartProcess = spawn(getNpmPath(), ['start'], { cwd: reactDir, timeout: 300000 });
-    // Give react server time to start
-    await new Promise((resolve) => setTimeout(resolve, 60000));
-    const res = execa.sync(getNpxPath(), ['cypress', 'run'], { cwd: reactDir, encoding: 'utf8' });
-    // kill the react server process
-    spawnSync('kill', [`${npmStartProcess.pid}`], { encoding: 'utf8' });
-    await new Promise((resolve) => setTimeout(resolve, 1000));
-    // Seriously, kill the react server process
-    // react-scripts somehow resurrects the process automatically after the first kill.
-    spawnSync('pkill', ['-f', 'react'], { encoding: 'utf8' });
-    expect(res.exitCode).toBe(0);
+    let npmStartProcess: ChildProcessWithoutNullStreams;
+    let res: execa.ExecaSyncReturnValue<string>;
+    try {
+      npmStartProcess = spawn(getNpmPath(), ['start'], { cwd: reactDir, timeout: 300000 });
+      // Give react server time to start
+      await new Promise((resolve) => setTimeout(resolve, 60000));
+      res = execa.sync(getNpxPath(), ['cypress', 'run'], { cwd: reactDir, encoding: 'utf8' });
+    } finally {
+      // kill the react server process
+      spawnSync('kill', [`${npmStartProcess.pid}`], { encoding: 'utf8' });
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      // Seriously, kill the react server process
+      // react-scripts somehow resurrects the process automatically after the first kill.
+      spawnSync('pkill', ['-f', 'react'], { encoding: 'utf8' });
+      expect(res.exitCode).toBe(0);
```

Comment:
should assertion go to line after `res = execa...` ??

Reply:
Actually it should not.

This test passes
```ts
describe('jest', () => {
  it('expect in try/catch', async () => {
    try {
      expect(1).toBe(2);
    } catch {
      // blank
    }
  });
});
```


but this does not
```ts
describe('jest', () => {
  it('expect in try/catch', async () => {
    try {
      expect(1).toBe(2);
    } catch {
      // blank
    } finally {
      expect(3).toBe(4);
    }
  });
});

```

Reply:
Yes. But... there's no `catch` block in the PR.

Reply:
Ah, that indeed changes the behavior, my mistake

Reply:
also, consider wrapping `it` in extra `describe` block and use after test handlers for cleanup.
`try-catch-finaly` (and variations) are quite fragile due to reasons you mentioned.